### PR TITLE
Add `path` arg to all commands

### PR DESCRIPTION
**Pull request by gvc**
## What's the purpose of this pull request?

To allow FastStore to be run from a monorepo we need to add this path argument so the FastStore CLI can locate the directory where the actual customizations are.

## How it works?

After https://github.com/vtex/faststore/2366 and https://github.com/vtex/faststore/2368 this PR just adds the arg to the OCLIF command class and pass it down to the code, to be used with `withBasePath`.

### CODE REVIEW

File: packages/core/src/server/generator/schema.ts [hunk 1]
```diff
@@ -68,7 +68,8 @@ export function writeGraphqlSchemaFile(apiSchema: GraphQLSchema) {
     // getting the schema before write because somehow this fixes the validation step of codegen from codesandbox
     const schema = printSchemaWithDirectives(apiSchema)
     writeFileSync(
-      path.join(process.cwd(), '@generated', 'schema.graphql'),
+      path.join(__dirname, '..', '..', '..', '@generated', 'schema.graphql'),
+      // path.join(process.cwd(), '@generated', 'schema.graphql'),
```
Comment (vaporwavie) [inline packages/core/src/server/generator/schema.ts:72]: @gvc are we keeping this commented for a further iteration?

Comment (gvc) [reply packages/core/src/server/generator/schema.ts:72]: I left this to see if you were paying attention to the review 🤣 

Sorry, I'll take it out :)

File: packages/core/src/server/generator/schema.ts [hunk 2]
```diff
@@ -68,7 +68,7 @@ export function writeGraphqlSchemaFile(apiSchema: GraphQLSchema) {
     // getting the schema before write because somehow this fixes the validation step of codegen from codesandbox
     const schema = printSchemaWithDirectives(apiSchema)
     writeFileSync(
-      path.join(process.cwd(), '@generated', 'schema.graphql'),
+      path.join(__dirname, '..', '..', '..', '@generated', 'schema.graphql'),
```
Comment (lariciamota) [inline packages/core/src/server/generator/schema.ts:71]: question: Is there a way to use that basePath here? I'm wondering if we could do something to make sure it would still work if something changed on the folders structure.

File: packages/cli/src/commands/test.ts [hunk 3]
```diff
@@ -42,8 +42,16 @@ async function storeTest(tmpDir: string) {
 }
 
 export default class Test extends Command {
+  static args = [
+    {
+      name: 'path',
+      description: 'The path where the FastStore being tested is. Defaults to cwd.',
+    }
```
Comment (lariciamota) [inline packages/cli/src/commands/test.ts:49]: question: isn't it necessary to add `required: false`?

File: packages/cli/src/commands/test.ts [hunk 4]
```diff
@@ -42,8 +42,16 @@ async function storeTest(tmpDir: string) {
...
+  ]
```
Comment (lariciamota) [inline packages/cli/src/commands/test.ts:50]: suggestion: What do you think about adding an example using `defaultHelp`?

File: packages/cli/src/commands/build.ts [hunk 5]
```diff
@@ -5,10 +5,21 @@ import { existsSync } from 'fs'
 import { copySync, moveSync, readdirSync, removeSync } from 'fs-extra'
 import { withBasePath } from '../utils/directory'
 import { generate } from '../utils/generate'
+import { ArgInput } from '@oclif/core/lib/interfaces'
```
Comment (lariciamota) [inline packages/cli/src/commands/build.ts:8]: question: why did you add this to this command, but not to the others?

File: packages/cli/src/utils/directory.test.ts [hunk 6]
```diff
@@ -11,18 +11,6 @@ const pathsToMatch = (expected: string, desired: string) => {
 describe('withBasePath as the current dir `.`', () => {
   const basePath = '.'
 
-  describe('coreDir', () => {
```
Comment (lariciamota) [inline packages/cli/src/utils/directory.test.ts:14]: question: why did you delete those tests?

File: packages/core/src/server/generator/schema.ts (hunk 2 above)
Comment (gvc) [reply packages/core/src/server/generator/schema.ts:71]: You're correct about not depending on folder structure. I'm unsure how to put the `basePath` (or a derived path) here since `core` is outside the scope of the CLI and has no access to `withBasePath`

File: packages/cli/src/commands/test.ts (hunk 3 above)
Comment (gvc) [reply packages/cli/src/commands/test.ts:49]: For this version of oclif/core the `required` is optional, which makes me think that if it's not there it will be assumed as `false`. I've checked this assumption and it holds.

Comment (gvc) [reply packages/cli/src/commands/test.ts:49]: This is the type definition: https://github.com/oclif/core/blob/v1.16.4/src/interfaces/parser.ts#L6-L15

File: packages/core/src/server/generator/schema.ts (hunk 2 above)
Comment (lariciamota) [reply packages/core/src/server/generator/schema.ts]: You're right... The only idea I came up now was to add a comment there with this warning, to make it more explicit that if this file change paths it will break. What do you think?

File: packages/cli/src/commands/build.ts (hunk 5 above)
Comment (gvc) [reply packages/cli/src/commands/build.ts]: By mistake! I've taken it out!

File: packages/cli/src/commands/test.ts (hunk 4 above)
Comment (gvc) [reply packages/cli/src/commands/test.ts:50]: The version of `oclif/core` being used does not have the `defaultHelp` argument 😰

File: packages/core/src/server/index.ts [hunk 7]
```diff
@@ -50,9 +50,12 @@ const formatError: FormatErrorHandler = (err) => {
 }
 
 function loadGeneratedSchema(): TypeSource {
-  return loadFilesSync(path.join(process.cwd(), '@generated'), {
-    extensions: ['graphql'],
-  })
+  return loadFilesSync(
+    path.join(process.cwd(), '@generated', 'schema.graphql'),
```
Comment (lariciamota) [inline packages/core/src/server/index.ts:54]: question: why did you add 'schema.graphql' here?

Comment (gvc) [reply packages/core/src/server/index.ts]: Undid this :) Was by mistake but there was no behavior change.

**Review by lariciamota — APPROVED** [review]

**Review by eduardoformiga — COMMENTED** [review]
The code LGTM.
Question: did you test using this cli version in the starter already?

**Review by eduardoformiga — APPROVED** [review]
I tested this version on the starter now and it worked as expected 🎉
